### PR TITLE
Add node.js app insights package

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -59,4 +59,12 @@ jobs:
...
       run: npm run lint
-      working-directory: sdk/feature-management-applicationinsights-browser
\ No newline at end of file
+      working-directory: sdk/feature-management-applicationinsights-browser
+
+    - name: Build feature-management-applicationinsights-node
+      run: npm run build
+      working-directory: sdk/feature-management-applicationinsights-node
+
+    - name: Run lint check for feature-management-applicationinsights-node
```
Comment (rossgrambo) [inline .github/workflows/ci.yml:68]: nit: should we lint before build? Just so it fails faster

File: sdk/feature-management-applicationinsights-node/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+    "env": {
+        "browser": true,
```
Comment (rossgrambo) [inline sdk/feature-management-applicationinsights-node/.eslintrc:3]: Should some of these be false for node?

File: sdk/feature-management-applicationinsights-node/src/telemetry.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ * @param client The Application Insights telemetry client.
+ * @returns A callback function that takes an evaluation result and tracks an event with the evaluation details.
+ */
+export function createTelemetryPublisher(client: TelemetryClient): (event: EvaluationResult) => void {
+    return (event: EvaluationResult) => {
+        if (event.feature === undefined) {
+            return;
+        }
+
+        const eventProperties = {
```
Comment (rossgrambo) [inline sdk/feature-management-applicationinsights-node/src/telemetry.ts:29]: Can we share this logic in the main library? Maybe just under Telemetry somehow?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (zhiyuanliang-ms) [reply .github/workflows/ci.yml:68]: Sounds good. Will create another pr to do this

File: sdk/feature-management-applicationinsights-node/.eslintrc (hunk 2 above)
Comment (zhiyuanliang-ms) [reply sdk/feature-management-applicationinsights-node/.eslintrc:3]: Interesting, I copied this file and didn't check the content. I think you are correct. Also update it for the browser package.

File: sdk/feature-management-applicationinsights-node/src/telemetry.ts (hunk 3 above)
Comment (zhiyuanliang-ms) [reply sdk/feature-management-applicationinsights-node/src/telemetry.ts:29]: Sounds good.

File: sdk/feature-management/src/telemetry/featureEvaluationEvent.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+export function createFeatureEvaluationEventProperties(result: EvaluationResult): any {
+    if (result.feature === undefined) {
+        return undefined;
+    }
+
+    const eventProperties = {
+        [VERSION]: EVALUATION_EVENT_VERSION,
+        [FEATURE_NAME]: result.feature ? result.feature.id : "",
+        [ENABLED]: result.enabled ? "True" : "False",
```
Comment (zhiyuanliang-ms) [inline sdk/feature-management/src/telemetry/featureEvaluationEvent.ts:24]: This change is to align with .NET and python FM

File: scripts/build-and-pack.sh [hunk 5]
```diff
@@ -41,4 +41,19 @@ echo "npm pack in $PACKAGE_DIR"
 npm pack
 
 echo "copy $PACKAGE package to $PROJECT_BASE_DIR"
-cp "$PACKAGE_DIR"/*.tgz "$PROJECT_BASE_DIR"
\ No newline at end of file
+cp "$PACKAGE_DIR"/*.tgz "$PROJECT_BASE_DIR"
```
Comment (rossgrambo) [inline scripts/build-and-pack.sh:44]: Should this .sh script lint test as well?

File: sdk/feature-management-applicationinsights-node/rollup.config.mjs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        format: "esm",
+        sourcemap: true,
+        preserveModules: true,
+      }
+    ],
+    plugins: [
+      typescript({
+        compilerOptions: {
+          "lib": [
+            "DOM",
```
Comment (rossgrambo) [inline sdk/feature-management-applicationinsights-node/rollup.config.mjs:26]: Do we need DOM in this node package?

**Review by rossgrambo — APPROVED** [review]

File: sdk/feature-management-applicationinsights-node/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{
+    "compilerOptions": {
+        "lib": [
+            "DOM",
```
[the remark this answers is not in the document]
Comment (zhiyuanliang-ms) [reply sdk/feature-management-applicationinsights-node/tsconfig.json:4]: Without this file, vscode may give you some red underscore.

File: sdk/feature-management-applicationinsights-node/rollup.config.mjs (hunk 6 above)
Comment (zhiyuanliang-ms) [reply sdk/feature-management-applicationinsights-node/rollup.config.mjs:26]: I reused the rollup.config left by @Eskibear . I guess this is from some template? Will figure it out later. But in this PR, I don't mind keeping them.

File: scripts/build-and-pack.sh (hunk 5 above)
Comment (zhiyuanliang-ms) [reply scripts/build-and-pack.sh:44]: I don't think it is necessary. Since our github ci pipeline will guard us.

**Pull request merged**